File: SDKs/googleapis.com/doubleclickbidmanager/v2/python/src/sdk/models/shared/runqueryrequest.py
import dataclasses
from typing import Optional
from dataclasses_json import dataclass_json
from sdk import utils
from ..shared import datarange as shared_datarange


@dataclass_json
@dataclasses.dataclass
class RunQueryRequest:
    r"""RunQueryRequest
    Request to run a stored query to generate a report.
    """
    
    data_range: Optional[shared_datarange.DataRange] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('dataRange') }})
    

File: SDKs/googleapis.com/doubleclickbidmanager/v2/java/lib/src/main/java/openapisdk/models/shared/RunQueryRequest.java
package openapisdk.models.shared;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonInclude.Include;

/**
 * RunQueryRequest
 * Request to run a stored query to generate a report.
**/
public class RunQueryRequest {
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("dataRange")
    public DataRange dataRange;
    public RunQueryRequest withDataRange(DataRange dataRange) {
        this.dataRange = dataRange;
        return this;
    }
}